### PR TITLE
Add route53 record set resource

### PR DESCRIPTION
**Pull request by moretea**

### CODE REVIEW

**Comment by lo1tuma** [comment]
This implements the suggested approach of a route53 RecordSet resource in #469, see also #630 and #527.
It doesn’t implement all route53 features yet but it would be very easy to add additional options in order to fix e.g. #174.
Using the route53 resource instead of the existing route53 functionality should also properly cleanup the DNS entries on destroy, see #147.

--------

A question I want raise is if it would make sense to document the deprecation of the old route53 functionality, so that it can be removed sometime in the future (next major release)?

File: nix/route53-recordset.nix [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+    name = mkOption {
+      default = "nixops-${uuid}-${name}";
+      type = types.str;
+      description = "Name of the recordset.";
+    };
+
+    region = mkOption {
+      type = types.str;
+      description = "AWS region.";
+    };
```
Comment (lo1tuma) [inline nix/route53-recordset.nix]: This option is currently not used as it is only required for latency-based resource record sets.

Comment (rbvermaa) [reply nix/route53-recordset.nix]: @lo1tuma If the option is not used, should it be removed?

Comment (moretea) [reply nix/route53-recordset.nix]: We've not implemented all potential behaviours for this resource. Just the ones they need at HolidayCheck. AFAICR other features do need to have a region specified.

**Comment by moretea** [comment]
I don't think that I will spend any time on this PR any time soonish, so feel free to change it any way you want ;)

File: nix/route53-recordset.nix (hunk 1 above)
Comment (lo1tuma) [reply nix/route53-recordset.nix]: As @moretea said, for the current feature set it is not needed. There are some route53 features that require a region. For now I think we can just remove it or at least make it optional.

**Comment by lo1tuma** [comment]
I’ve added another [comment to #469](https://github.com/NixOS/nixops/pull/469#issuecomment-340769909) because I think having route53 as a separate resource cannot solve the use-case I have.
So I guess we need some more thinking and discussions in how nixops should handle route53.

**Comment by rbvermaa** [comment]
I am going to work on route53 support the next weeks, and will likely include your commit in the work and improve if necessary.

**Comment by moretea** [comment]
Awesome :)

**Comment by rbvermaa** [comment]
@lo1tuma The work I am doing should include your use-case btw. The most likely solution will be allowing machine resources to be used in the recordValue, with the option of using either public or private IP address of the instance/machine.

**Comment by rbvermaa** [comment]
Closing this PR. I have used the commit in the route53 branch which should be merged soon.

**Pull request closed without merging**